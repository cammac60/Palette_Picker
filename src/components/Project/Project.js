import React, { useState, useEffect } from 'react';
import { Link } from 'react-router-dom';
import { BrowserRouter } from 'react-router-dom';
import { getPalettes, removeItem } from '../../apiCalls';
import './Project.scss';

export const handleDelete = async (id, item, hooks) => {
  const response = await removeItem(id, item);
  const updatedPalettes = hooks.palettes.filter(palette => palette.id !== id);
  hooks.setPalettes(updatedPalettes);
}

export const Project = props => {
  const { id, user, setUser, projects, setProjects } = props;
  const [palettes, setPalettes] = useState([]);
  const removeItemHooks = {palettes, setPalettes, projects, setProjects}

  useEffect(() => {
    const fetchPalettes = async () => {
      try {
        const response = await getPalettes(id);
        setPalettes(response.palettes);
      } catch(error) {
          console.log(error);
      }
    };
    fetchPalettes();
  }, []);

  const formatPalettes = palettes.map(pal => {
      console.log('palette.name', pal.name);
      return (
          <section className='proj-palette' key={pal.id} id={pal.id}>
            <h3 className='proj-palette-name'>{pal.name}</h3>
            <div className='color-box-container'>
              <div className='color_box' style={{backgroundColor: pal.color_one}}></div>
              <div className='color_box' style={{backgroundColor: pal.color_two}}></div>
              <div className='color_box' style={{backgroundColor: pal.color_three}}></div>
              <div className='color_box' style={{backgroundColor: pal.color_four}}></div>
              <div className='color_box' style={{backgroundColor: pal.color_five}}></div>
            </div>
            <img alt="Trash Icon" className="trash-icon"
              src={require("../../images/icons/delete.svg")}
              onClick={() => handleDelete(pal.id, 'palette', removeItemHooks)}/>
          </section>
        )
      });

  return (
    <div className='project-page'>
      <nav className="project-nav">
        <h2>Palette Picker</h2>
        <BrowserRouter>
        <Link className="back-button" to="/">Back </Link>
        <Link className="sign-out" to="/" onClick={() => setUser({})}>Sign Out</Link>
        </BrowserRouter>
      </nav>
      <div className='project-div'>
        <h2>Project {id}</h2>
        <button className='delete-project-button'>Delete Project</button>
      </div>
      <div className='palettes-container'>
        {palettes.length ? formatPalettes : 'Loading'}
      </div>
    </div>
  )
}

export default Project;
