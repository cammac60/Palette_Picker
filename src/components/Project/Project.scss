@import '../../variables.scss';

.back-button {
  color: $light-text-color;
  font-size: 1.2rem;
  margin: {
    bottom: auto;
    top: auto;
  }
  &:hover {
    color: $medium-text-color;
    text-decoration: none;
  }
}
