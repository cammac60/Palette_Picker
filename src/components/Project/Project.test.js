import React, {useState} from 'react';
import { Project } from './Project.js';
import { shallow } from 'enzyme';

describe('Project', () => {
  let wrapper, mockProjects;

  beforeEach(() => {
    mockProjects = [
        {id: 1, name: 'Project 1'},
        {id: 2, name: 'Project 2'}
      ];
  });

  it('Should match snapshot when no palettes exist for a project', () => {
    wrapper = shallow(<Project
      projects={mockProjects}
    />)
    expect(wrapper).toMatchSnapshot();
  });
  
});
