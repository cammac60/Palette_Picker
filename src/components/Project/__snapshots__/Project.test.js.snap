// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project Should match snapshot when no palettes exist for a project 1`] = `
<div
  className="project-page"
>
  <nav
    className="project-nav"
  >
    <h2>
      Palette Picker
    </h2>
    <div
      className="nav-buttons"
    >
      <Link
        className="back-arrow"
        to="/home"
      >
        <img
          alt="Back Arrow Icon"
          className="back-arrow-icon"
          src={
            Object {
              "ReactComponent": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
              "__esModule": true,
              "default": "down-arrow.svg",
            }
          }
        />
      </Link>
      <Link
        className="back-button"
        to="/home"
      >
        Back 
      </Link>
      <Link
        className="sign-out"
        onClick={[Function]}
        to="/"
      >
        Sign Out
      </Link>
    </div>
  </nav>
  <div
    className="project-div"
  >
    <h2 />
    <button
      className="delete-project-button"
      onClick={[Function]}
    >
      Delete Project
    </button>
  </div>
  <h3>
    No palettes yet!
  </h3>
  <div
    className="palettes-container"
  />
</div>
`;
